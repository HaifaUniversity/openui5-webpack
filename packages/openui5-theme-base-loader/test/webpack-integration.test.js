const fs = require('fs');
const path = require('path');
const webpack = require('webpack');

const cases = fs.readdirSync(path.join(__dirname, 'cases'));
describe('Integration TestCases', () => {
  cases.forEach((testCase) => {
    it(testCase, (done) => {
      const options = {
        entry: {
          build: './index.js',
        },
        resolve: {
          modules: [
            "test_components/test-library"
          ]
        },
        resolveLoader: {
          alias: {
            'openui5-theme-base-loader': path.join(__dirname, "../src/index")
          }
        },
        module: {
          rules: [
            {
              test: /\.less$/,
              use: 'less-loader'
            },
          ],
        },
        optimization: {
          minimize: false,
          runtimeChunk: 'single',
        },
        mode: 'production'
      };

      const testDirectory = path.join(__dirname, 'cases', testCase);
      const outputDirectory = path.join(__dirname, 'js', testCase);

      options.context = testDirectory;
      options.output = { filename: '[name].js' };
      options.output.path = outputDirectory;

      webpack(options, (err, stats) => {
        if (err) {
          done(err);
          return;
        }
        if (stats.hasErrors()) {
          done(new Error(stats.toString()));
          return;
        }
        const actualPath = path.join(outputDirectory, 'build.js');
        expect(readFileOrEmpty(actualPath)).toMatchSnapshot();
        done();
      });
    });
  });
});

function readFileOrEmpty(path) {
  try {
    return fs.readFileSync(path, 'utf-8');
  } catch (e) {
    return '';
  }
}
