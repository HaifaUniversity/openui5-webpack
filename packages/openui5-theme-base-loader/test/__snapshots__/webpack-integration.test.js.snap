// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration TestCases add-dependency 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[1],[
/* 0 */
/***/ (function(module, exports) {



/***/ }),
/* 1 */
/***/ (function(module, exports, __webpack_require__) {

__webpack_require__(0);


/***/ })
],[[1,0]]]);"
`;
